### PR TITLE
+pnpm.io

### PR DESCRIPTION
**Pull request by getneil**
[pnpm](https://pnpm.io/) a better and faster node package manager

### CODE REVIEW

**Comment by mxcl** [comment]
Like our packaging guide recommends, see what brew do: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pnpm.rb

File: projects/pnpm.io/package.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+distributable:
+  url: https://github.com/pnpm/pnpm/archive/refs/tags/v{{ version }}.tar.gz
```
Comment (mxcl) [inline projects/pnpm.io/package.yml]: use: `https://registry.npmjs.org/pnpm/-/pnpm-{{version}}.tgz` per brew formula

File: projects/pnpm.io/package.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+dependencies:
+  nodejs.org: '>=16'
+
+build:
+  dependencies:
+    curl.se: '>=5'
+    nodejs.org: '>=16'
+  script:
+    curl -L https://get.pnpm.io/v6.16.js | node - add --global pnpm@next-7 npm@7
```
Comment (mxcl) [inline projects/pnpm.io/package.yml]: Seems like the tarball is in a ready to execute format: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pnpm.rb#L29-L33

File: projects/pnpm.io/package.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+provides:
+  - bin/pnpm
+
+dependencies:
+  nodejs.org: '>=16'
+
+build:
+  dependencies:
+    curl.se: '>=5'
```
Comment (mxcl) [inline projects/pnpm.io/package.yml]: don't forget to drop this

File: projects/pnpm.io/package.yml (hunk 2 above)
Comment (getneil) [reply projects/pnpm.io/package.yml]: thank you for this big help.

File: projects/pnpm.io/package.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+dependencies:
+  nodejs.org: '>=16'
+  npmjs.com: '>=8'
+
+build:
+  script: |
+    npm install
+    export PNPM_HOME=root/Library/pnpm
+    export PATH="$PNPM_HOME:$PATH";
+    export PATH="$(pwd)/bin:$PATH";
```
Comment (mxcl) [inline projects/pnpm.io/package.yml]: your script doesn't move anything to `{{prefix}}` so nothing is installed. Confused what you're doing anyway, all brew does is move a few files: https://github.com/Homebrew/homebrew-core/blob/master/Formula/pnpm.rb#L29-L33

Comment (getneil) [reply projects/pnpm.io/package.yml]: sorry did not fully understand the purpose of {{prefix}}, i just wrongly assumed pwd would be where to just put stuff. but thanks im finally getting a different error now.

File: projects/pnpm.io/package.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+distributable:
+  url: https://registry.npmjs.org/pnpm/-/pnpm-{{version}}.tgz
+  strip-components: 1
+
+versions:
+  github: pnpm/pnpm
```
Comment (getneil) [inline projects/pnpm.io/package.yml:6]: dont know yet, but i feel im missing something here... hmmmm

File: projects/pnpm.io/package.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+distributable:
+  # url: https://registry.npmjs.org/pnpm/-/pnpm-{{version}}.tgz
+  url: https://registry.npmjs.org/pnpm/-/pnpm-7.18.2.tgz
```
Comment (getneil) [inline projects/pnpm.io/package.yml]: i think pnpm is returning a bunch of weird versions with chars.

**Comment by mxcl** [comment]
Excellent!

**Comment by mxcl** [comment]
I do wonder if you were building locally first? That way is much faster than using CI/CD to evaluate build steps.

**Comment by getneil** [comment]
> I do wonder if you were building locally first? That way is much faster than using CI/CD to evaluate build steps.

i tried first trying to run it directly failed on that, tried to dockerize also (encountered setting up linux on m1)
then i tried to run the entire pipeline using https://github.com/nektos/act the only part that run was the `get-diff`
then I tried using my mac mini intel there some parts started working correctly for the linux part

**Comment by mxcl** [comment]
what did you try? What failed?

If you have issues with tea please tell us about them here or in Slack, if you are having issues then hundred other people also are and they also aren't reporting them.

**Pull request merged**